### PR TITLE
Create a proper directory for each snapshot 

### PR DESCRIPTION
**Pull request by jorisdral**
And put the snapshot metadata file in there. In addition, compute the checksum for the metadata file and put it in a separate file in the same directory.

### CODE REVIEW

**Comment by wenkokke** [comment]
LGTM

File: src/Database/LSMTree/Internal.hs [hunk 1]
```diff
@@ -1105,9 +1105,17 @@ snapshot resolve snap label tableType t = do
...
-              when b $ throwIO (ErrSnapshotExists snap)
+
+      -- Guard that the snapshot does not exist already
+      let snapDir = Paths.namedSnapshotDir (tableSessionRoot thEnv) snap
+      doesSnapshotExist <-
+        FS.doesDirectoryExist (tableHasFS thEnv) (Paths.getNamedSnapshotDir snapDir)
+      if doesSnapshotExist then
+        throwIO (ErrSnapshotExists snap)
+      else
+        -- we assume the snapshots directory already exists, so we just have to
```
Comment (wenkokke) [inline src/Database/LSMTree/Internal.hs:1116]: Where is the snapshots directory created? Should we handle the error when it does not exist?

File: src/Database/LSMTree/Internal.hs [hunk 2]
```diff
@@ -1105,9 +1105,17 @@ snapshot resolve snap label tableType t = do
...
-      FS.doesFileExist (tableHasFS thEnv) snapPath >>= \b ->
-              when b $ throwIO (ErrSnapshotExists snap)
+
+      -- Guard that the snapshot does not exist already
+      let snapDir = Paths.namedSnapshotDir (tableSessionRoot thEnv) snap
+      doesSnapshotExist <-
+        FS.doesDirectoryExist (tableHasFS thEnv) (Paths.getNamedSnapshotDir snapDir)
+      if doesSnapshotExist then
+        throwIO (ErrSnapshotExists snap)
+      else
```
Comment (wenkokke) [inline src/Database/LSMTree/Internal.hs:1115]: Should we extract this into a helper function? It feels like it's repeated quite a bit throughout.

Comment (jorisdral) [reply src/Database/LSMTree/Internal.hs:1115]: Hmm, I tried to provide helper functions but it's not simplifying the code much, so I'm keeping it as is

Comment (wenkokke) [reply src/Database/LSMTree/Internal.hs:1087]: Seems fine. It is quite simple.

**Review by wenkokke — APPROVED** [review]

**Pull request merged**